### PR TITLE
Add `GDT_Int8`, test GDAL 3.7 on CI and refactor workflow

### LINKED ISSUE #403: Create CI builds against GDAL 3.6 & 3.7
Should we consider dropping 3.2 and/or 3.3? Under what principles should we make that evaluation?

### PR DESCRIPTION
**Pull request by lnicola**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Closes #403

### CODE REVIEW

**Comment by lnicola** [comment]
bors try

File: bors.toml [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
-    "ci gdal-33",
-    "ci gdal-33 (--all-features)",
-    "ci gdal-32",
-    "ci gdal-32 (--all-features)",
+    "ci gdal-3.7",
+    "ci gdal-3.6",
+    "ci gdal-3.5",
+    "ci gdal-3.4",
+    "ci gdal-3.3",
+    "ci gdal-3.2",
```
Comment (metasim) [inline bors.toml:7]: OMG ❤️

**Review by metasim — APPROVED** [review]

**Comment by metasim** [comment]
@lnicola 🙇 🙇 🙇

**Pull request merged**